core/src/main/java/org/apache/sedona/core/io/avro/utils/AvroUtils.java

```diff
@@ -0,0 +1,263 @@
+                               IntStream.range(0, polygon.getNumInteriorRing())
+                                        .mapToObj(polygon::getInteriorRingN)
+                                        .map(linearRing -> getCollectionFromCoordinates(linearRing.getCoordinates()))
+                                        .collect(Collectors.toList()));
+    }
+    
+    public static GenericRecord getRecordFromGeometry(Schema schema,
+                                                      String geometryColumnName,
+                                                      String userColumn,
+                                                      Object geometryData,
+                                                      Object userData) throws SedonaException {
+        return (GenericRecord) getRecord(schema,
+                                         ImmutableMap.of(geometryColumnName, geometryData, userColumn, userData));
+    }
+    
+    public static GenericRecord getRecordFromGeometry(Schema schema,
+                                                      String geometryColumnName,
+                                                      Object geometryData,
+                                                      Optional<Map<String, Object>> userData) throws SedonaException {
+        return (GenericRecord) getRecord(schema,
+                                         (Object) new ImmutableMap.Builder<>().putAll(userData.orElse(Collections.EMPTY_MAP))
+                                                                              .put(geometryColumnName, geometryData)
+                                                                              .build());
+    }
+    
+    public static Schema getSchema(GeometryType geometryType,
+                                   String geometryColumnName,
+                                   List<Field> userColumns,
+                                   String namespace,
+                                   String name) throws SedonaException {
+        Field geometryField = getField(geometryColumnName, geometryType);
+        return getGeometryRecordSchema(namespace, name, geometryField, userColumns);
+    }
+    
+    public static GenericRecord getRecord(Geometry geometry,
+                                          GeometryType type,
+                                          String geometryColumnName,
+                                          Schema avroSchema) throws SedonaException {
+        Object geometryData;
+        switch (type) {
+            case CIRCLE: {
+                geometryData = getMapFromCircle((Circle) geometry);
+                break;
+            }
+            case LINESTRING: {
+                geometryData = getCollectionFromLineString((LineString) geometry);
+                break;
+            }
+            case POINT: {
+                geometryData = getMapFromPoint((Point) geometry);
+                break;
+            }
+            case POLYGON:
```

Comment:
What about MultiPolygons? Wouldn`t it be great if complex geometries are supported natively?